sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts

```diff
@@ -102,73 +135,277 @@ export class StudentScholasticStandingsService extends RecordDataModelService<St
         .getOne();
 
-      // Cloning existing offering.
-      const offering: EducationProgramOffering = { ...existingOffering };
-
-      // Assigning id as undefined, so that when its saved its considered as a new EducationProgramOffering object.
-      offering.id = undefined;
-
-      const newStudyEndDate =
-        scholasticStandingData.dateOfChange ??
-        scholasticStandingData.dateOfCompletion ??
-        scholasticStandingData.dateOfIncompletion ??
-        scholasticStandingData.dateOfWithdrawal;
-
-      offering.studyEndDate = new Date(newStudyEndDate);
-
-      offering.actualTuitionCosts =
-        scholasticStandingData.tuition ?? existingOffering.actualTuitionCosts;
-      offering.programRelatedCosts =
-        scholasticStandingData.booksAndSupplies ??
-        existingOffering.programRelatedCosts;
-      offering.mandatoryFees =
-        scholasticStandingData.mandatoryFees ?? existingOffering.mandatoryFees;
-      offering.exceptionalExpenses =
-        scholasticStandingData.exceptionalCosts ??
-        existingOffering.exceptionalExpenses;
-
-      offering.offeringType = OfferingTypes.ScholasticStanding;
-
-      // Save new offering.
-      const savedOffering = await transactionalEntityManager
-        .getRepository(EducationProgramOffering)
-        .save(offering);
+      // Check for restrictions and apply if any.
+      const studentRestriction = await this.getScholasticStandingRestrictions(
+        scholasticStandingData,
+        existingOffering.offeringIntensity,
+        application.studentId,
+        auditUserId,
+        application.id,
+      );
+      if (studentRestriction) {
+        await transactionalEntityManager
+          .getRepository(StudentRestriction)
+          .save(studentRestriction);
+      }
 
       // Create StudentScholasticStanding.
       const scholasticStanding = new StudentScholasticStanding();
-      scholasticStanding.application = { id: applicationId } as Application;
+      scholasticStanding.application = { id: application.id } as Application;
       scholasticStanding.submittedData = scholasticStandingData;
       scholasticStanding.submittedDate = now;
       scholasticStanding.submittedBy = auditUser;
       scholasticStanding.creator = auditUser;
 
-      // Create the new assessment to be processed.
-      scholasticStanding.studentAssessment = {
-        application: { id: applicationId } as Application,
-        triggerType: AssessmentTriggerType.ScholasticStandingChange,
-        creator: auditUser,
-        submittedBy: auditUser,
-        submittedDate: now,
-        offering: { id: savedOffering.id } as EducationProgramOffering,
-      } as StudentAssessment;
+      // Reference offering id.
+      scholasticStanding.referenceOffering = existingOffering;
+
+      // If not unsuccessful weeks, then clone new offering and create re-assessment.
+      const notUnsuccessfulWeeksScholasticStanding =
```

Comment:
Please remove unnecessary conditions. `scholasticStandingData.numberOfUnsuccessfulWeeks` is not necessary here.